### ISSUE TITLE
Break Method Convert

### ISSUE THREAD

**Issue by Liaojinghui** [neo-project/neo-devpack-dotnet#816]
As is in https://github.com/neo-project/neo-devpack-dotnet/pull/813, the methodconvert file is too large to maintain, we need to break it into smaller files.

This issue will trace the breaking prs. Breaking process plan:
- [ ] ExpressionConvert https://github.com/neo-project/neo-devpack-dotnet/pull/818
- [ ] PatternConvert
- [ ] StatementConvert
- [ ] SystemConvert
- [ ] ConstructorConvert
- [ ] Adding comments to ExpressionConvert
- [ ] Adding comments to PatternConvert
- [ ] Adding comments to StatementConvert
- [ ] Adding comments to SystemConvert
- [ ] Adding comments to ConstructorConvert
- [ ] Adding tests to ExpressionConvert
- [ ] Adding tests to PatternConvert
- [ ] Adding tests to StatementConvert
- [ ] Adding tests to SystemConvert
- [ ] Adding tests to ConstructorConvert

**Comment by shargon** [comment]
Could you add unit tests at same time that is moved?

**Comment by Liaojinghui** [comment]
Work all at once would only cause two situations, either very large pr again, or very surface unit tests. Unit tests for this has being missing for years, adding them back should be another task.

**Comment by shargon** [comment]
You say that splitting this method into several files is necessary to maintain it better and improve the coverage, I think that if you split it up it has to be to add that coverage, otherwise, we don't improve anything. In addition, this way we can ensure that this code is correct.

**Comment by Liaojinghui** [comment]
@shargon step by step. How about we focus on fixing issues? Large file is hard to follow and maintain, how about we break it into smaller files step by step, then  we work on checking C# language specification to find what is missing, then unit tests to ensure everything works fine, and then comments to make sure others can understand?

**Comment by Liaojinghui** [comment]
Will directly work on the original methodconvert file
